packages/react-components/src/components/InputBoxComponent.tsx

```diff
@@ -589,45 +445,70 @@ export const InputBoxComponent = (props: InputBoxComponentProps): JSX.Element =>
           }
         }
       }
-      let result = '';
-      if (tagsValue.length === 0) {
-        // no tags in the string and newValue should be used as a result string
-        result = newValue;
-      } else {
-        // there are tags in the text value and htmlTextValue is html string
-        // find diff between old and new text
-        const { changeStart, oldChangeEnd, newChangeEnd } = findStringsDiffIndexes({
-          oldText: inputTextValue,
-          newText: newValue,
-          previousSelectionStart: previousSelectionStartValue,
-          previousSelectionEnd: previousSelectionEndValue,
-          currentSelectionStart: currentSelectionStartValue,
-          currentSelectionEnd: currentSelectionEndValue
-        });
-        const change = newValue.substring(changeStart, newChangeEnd);
-        // get updated html string
-        const updatedContent = updateHTML({
-          htmlText: htmlTextValue,
-          oldPlainText: inputTextValue,
-          newPlainText: newValue,
-          tags: tagsValue,
-          startIndex: changeStart,
-          oldPlainTextEndIndex: oldChangeEnd,
-          change,
-          mentionTrigger: triggerText
-        });
-        result = updatedContent.updatedHTML;
-        // update caret index if needed
-        if (updatedContent.updatedSelectionIndex !== null) {
-          setCaretIndex(updatedContent.updatedSelectionIndex);
-          setSelectionEndValue(updatedContent.updatedSelectionIndex);
-          setSelectionStartValue(updatedContent.updatedSelectionIndex);
-        }
+    },
+    [debouncedQueryUpdate, mentionLookupOptions, updateMentionSuggestions]
+  );
+
+  /* @conditional-compile-remove(mention) */
+  const handleOnChange = useCallback(
+    async (
+      event: React.FormEvent<HTMLInputElement | HTMLTextAreaElement>,
+      tagsValue: TagData[],
+      htmlTextValue: string,
+      inputTextValue: string,
+      currentTriggerStartIndex: number,
+      previousSelectionStart?: number,
+      previousSelectionEnd?: number,
+      currentSelectionStart?: number,
+      currentSelectionEnd?: number,
+      updatedValue?: string
+    ): Promise<void> => {
+      debouncedQueryUpdate.cancel();
+      if (event.currentTarget === null) {
+        return;
       }
+      // handle backspace change
+      // onSelect is not called for backspace as selection is not changed and local caret index is outdated
+      setCaretIndex(undefined);
+      const newValue = updatedValue ?? '';
```

Comment:
Sometimes after deleting the whole string from the send box and starting to input "Hi @patricia Adams and" lead that the last symbol is inputted instead of the real keys that are selected on the keyboard. So instead of "Hi @patricia Adams and", I see "Hi @patricia Adams aaa"